src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBroker.cs

```diff
@@ -225,7 +226,12 @@ internal static void SetBrokerResult(Intent data, int resultCode, ICoreLogger un
                         var httpResponse = new HttpResponse();
                         httpResponse.Body = errorResult[BrokerResponseConst.BrokerHttpBody];
                         httpResponse.Headers = errorResult[BrokerResponseConst.BrokerHttpHeaders];
-                        httpResponse.StatusCode = errorResult[BrokerResponseConst.BrokerHttpStatusCode];
+
+                        HttpStatusCode status;
+                        if (Enum.TryParse(errorResult[BrokerResponseConst.BrokerHttpStatusCode], out status))
```

Comment:
Not sure what's happening here really. This is why "dynamic" should be avoided - no way of knowing what the objects really are. Maybe we should look again at why normal JSON parsing is failing in this case and avoid using dynamic.

Ultimately, I would only expect there to be strings when parsing json, not sure where there is an enum...